src/Mutex/Impl/Internal/MutexWorkflow.workflow.cs

```diff
@@ -0,0 +1,55 @@
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+[Workflow]
+internal class MutexWorkflow
+{
+    private readonly ILockHandler lockHandler = WorkflowMutex.CreateLockHandler();
+    private readonly Queue<LockRequest> requests = new();
+
+    [WorkflowRun]
+    public async Task RunAsync(MutexWorkflowInput? input = null)
```

Comment:
Often clearer to accept the non-null input and make your caller just pass an empty form of it. You're the only caller of this, so you can pass an empty input.